pkg/apis/seedmanagement/types_managedseedset.go

```diff
@@ -0,0 +1,128 @@
+	// Template / ShootTemplate.
+	UpdateStrategy *ManagedSeedSetUpdateStrategy
+	// RevisionHistoryLimit is the maximum number of revisions that will
+	// be maintained in the ManagedSeedSet's revision history. Defaults to 10.
+	RevisionHistoryLimit *int32
+}
+
+// ManagedSeedSetUpdateStrategy specifies the strategy that the ManagedSeedSet
+// controller will use to perform updates. It includes any additional parameters
+// necessary to perform the update for the indicated strategy.
+type ManagedSeedSetUpdateStrategy struct {
+	// Type indicates the type of the ManagedSeedSetUpdateStrategy. Defaults to ManagedSeedSetUpdateStrategyType.
+	Type *ManagedSeedSetUpdateStrategyType
+	// RollingUpdate is used to communicate parameters when Type is ManagedSeedSetUpdateStrategyType.
+	RollingUpdate *RollingUpdateManagedSeedSetUpdateStrategy
+}
+
+// ManagedSeedSetUpdateStrategyType is a string enumeration type that enumerates
+// all possible update strategies for the ManagedSeedSet controller.
+type ManagedSeedSetUpdateStrategyType string
+
+const (
+	// RollingUpdateManagedSeedSetUpdateStrategyType indicates that update will be
+	// applied to all ManagedSeeds / Shoots in the ManagedSeedSet with respect to the ManagedSeedSet
+	// ordering constraints.
+	RollingUpdateManagedSeedSetUpdateStrategyType ManagedSeedSetUpdateStrategyType = "RollingUpdate"
+)
+
+// RollingUpdateManagedSeedSetStrategy is used to communicate parameter for RollingUpdateManagedSeedSetUpdateStrategyType.
+type RollingUpdateManagedSeedSetUpdateStrategy struct {
+	// Partition indicates the ordinal at which the ManagedSeedSet should be partitioned. Defaults to 0.
+	Partition *int32
+}
+
+// ManagedSeedSetStatus represents the current state of a ManagedSeedSet.
+type ManagedSeedSetStatus struct {
+	// ObservedGeneration is the most recent generation observed for this ManagedSeedSet. It corresponds to the
+	// ManagedSeedSet's generation, which is updated on mutation by the API Server.
+	ObservedGeneration int64
+	// Replicas is the number of replicas (ManagedSeeds and their corresponding Shoots) created by the ManagedSeedSet controller.
+	Replicas int32
+	// ReadyReplicas is the number of ManagedSeeds created by the ManagedSeedSet controller that have a Ready Condition.
+	ReadyReplicas int32
+	// NextReplicaNumber is the ordinal number that will be assigned to the next replica of the ManagedSeedSet.
+	NextReplicaNumber int32
+	// CurrentReplicas is the number of ManagedSeeds created by the ManagedSeedSet controller from the ManagedSeedSet version
+	// indicated by CurrentRevision.
+	CurrentReplicas int32
+	// UpdatedReplicas is the number of ManagedSeeds created by the ManagedSeedSet controller from the ManagedSeedSet version
+	// indicated by UpdateRevision.
+	UpdatedReplicas int32
+	// CurrentRevision, if not empty, indicates the version of the ManagedSeedSet used to generate ManagedSeeds with smaller
+	// ordinal numbers during updates.
```

Comment:
The description in the StatefulSet doesn't not fit unfortunately. The StatefulSet always has replicas with ordinals from 0 to `replicas` - 1, so it's possible to specify the 2 ranges precisely. In our case, the ordinal numbers are ever increasing, so it's not possible to specify these ranges precisely. There will be `CurrentReplicas` replicas on `CurrentRevision` (with smaller ordinal numbers), `UpdatedReplicas` on `UpdatedRevision` (with larger ordinal numbers), and a "cutoff" number (or range) somewhere in the middle separating both ranges, but it's impossible to precisely tell the cutoff value (the controller will calculate it each time after enumerating all replicas). I chose to write a shorter comment rather than explaining all this. If you have an idea how to put it more concisely, please share.

Reply:
thanks. No, it is fine with me how it is currently.